### DISCUSSION TITLE
Live Update: bundle content?

### DISCUSSION THREAD

**Discussion by Aymkdn** [capawesome-team/capacitor-plugins#238] [Q&A]
Hey,

I'm interested by the live-update plugin. However I'm not sure what to put in the bundle Zip file? Should it be the content of my `dist` folder (I'm using Vue/Vite to produce it)?

So, an example of the bundle content should be the below?
  - index.html
  - favicon.ico
  - assets/
  - assets/index.xyz.js
  - assets/image.png
  - assets/web.abc.js

**Comment by robingenz** [comment]
Yes, exactly. You must upload your zipped dist folder.

**Reply by osamahkenawy** [reply]
Correct but check for configurations some platforms might require additional configuration files json manifest env or any of configuration files

**Reply by robingenz** [reply]
What do you mean exactly? Do you have an example? You can only update the web part of your app using live updates. Feel free to check out [this guide](https://capawesome.io/blog/announcing-the-capacitor-live-update-plugin/).

**Comment by domlysi** [comment]
In my case, I'm using Capacitor with the Quasar Framework.
Do I need to take the "quasar build" produced code and zip it? In other words, what's the requirements / schema for the bundle?

**Reply by robingenz** [reply]
Please take a look at [Create a bundle](https://capawesome.io/cloud/live-updates/bundles/#create-a-bundle):

> In order to create a bundle (aka "deploy a live update") on the Capawesome Cloud, you must first create the bundle locally by compiling your web application. In Angular, for example, you can bundle your app by running ng build. This will create a dist or www folder containing the compiled web assets. Each bundle must contain an index.html file at the root level of the folder.

I am not familiar with the Quasar framework but from the [documentation](https://quasar.dev/quasar-cli-webpack/developing-capacitor-apps/build-commands) I can see that `quasar build` should be the correct command. So you need to zip the folder that contains the `index.html` file of the web build output.

**Reply by domlysi** [reply]
Thank you for your advice. I'm trying to build it without the cloud. When I use quasar build it downloads and updates the app with the bundle, but on reload it says "Webpage not available - https://localhost could not be loaded. ERR_CONNECTION_REFUSED".

**Reply by robingenz** [reply]
Feel free to drop your bundle here and i will take a look.

**Reply by domlysi** [reply]
Thanks for looking into it!

[6.zip](https://github.com/user-attachments/files/xxxxxx)

And this is my current implementation:


```javascript
 async function updateApp() {
    const {LiveUpdate} = await import('@capawesome/capacitor-live-update')
    const {bundleIds} = await LiveUpdate.getBundles()
    api.get('app/update/').then(async (resp) => {
      if (resp.status !== 200) {
        console.log('Could not check for updates', resp.status)
        return
      }
      const {bundle_id, url} = resp.data

      if (bundleIds.includes(bundle_id)) {
        console.info('Bundle is up-to-date', bundle_id)
        return
      }

      console.info('downloading', bundle_id)

      let isDownloaded = false
      try {
        await LiveUpdate.downloadBundle({
          url: url,
          bundleId: bundle_id,
        })
        console.debug('Bundle downloaded')
        isDownloaded = true
      } catch (err) {
        console.error('Could not download bundle', err)
      }

      if (isDownloaded) {
        await LiveUpdate.reload()
      }
      await LiveUpdate.ready()
    })
  }

```

**Reply by robingenz** [reply]
Your bundle looks good.

> When I use quasar build it downloads and updates the app with the bundle, but on reload it says "Webpage not available - [localhost](https://localhost/) could not be loaded. ERR_CONNECTION_REFUSED".

Can you share a screenshot?

**Reply by domlysi** [reply]
I finally figured it out. So this is what I was doing wrong:

In Quasar you can develop for multiple platforms including web. There is an intern function "$q.platform.is.nativeMobile" that I used a lot in my code to differentiate between web and ios/android.

The Problem was building the bundle with `quasar build`, because it will not set **$q.platform.is.nativeMobile** to `true` and therefore there will be Errors mostly because some Settings are incorrect.

Building with `quasar build -T android -m capactior --ide` created a capacitor folder with the correct www files, that I zipped.

It seems to be a working but hacky solution. :)

Thanks for looking into it, tho. 

Edit: Maybe a more advanced example (and a full example) and docs would help to make the plugin easier to use. "Announcing the Capacitor Live Update Plugin" was already helpful, tho :+1:
